apps/explorer/tests/test_views.py

```diff
@@ -50,3 +50,230 @@ def test_renders_pixelset_list(self):
             '<tr class="pixelset">',
             count=10
         )
+
+    def test_species_filter(self):
+
+        # Create 8 pixelset
+        make_development_fixtures(
+            n_pixel_sets=8,
+            n_pixels_per_set=1
+        )
+
+        # Create our special ones
+        species = factories.SpeciesFactory()
+        strain = factories.StrainFactory(species=species)
+        n_new_pixel_sets = 2
+        pixel_sets = factories.PixelSetFactory.create_batch(n_new_pixel_sets)
+
+        omics_units = factories.OmicsUnitFactory.create_batch(
+            n_new_pixel_sets,
+            strain=strain
+        )
+        for omics_unit, pixel_set in zip(omics_units, pixel_sets):
+            factories.PixelFactory(
+                omics_unit=omics_unit,
+                pixel_set=pixel_set,
+            )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=10
+        )
+
+        # species filter
+        data = {
+            'species': [species.id, ]
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=n_new_pixel_sets
+        )
+
+    def test_omics_unit_types_filter(self):
+
+        # We generate one pixel per pixelset so that half of pixel sets
+        # contains pixels with 'mRNA' omics unit type and the other half
+        # 'protein' omics unit type
+        n_pixel_sets = 8
+        make_development_fixtures(
+            n_pixel_sets=n_pixel_sets,
+            n_pixels_per_set=1
+        )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
```

Comment:
This seems a bit weak. Can't you pass a CSS element instead?

Reply:
Do you mean a CSS selector like `tr.pixelset`?

Reply:
Or only `.pixelset`

Reply:
Unfortunately no. But, note that the assertion is in "HTML mode", it's a little bit weaker. See the [documentation](https://docs.djangoproject.com/en/1.11/topics/testing/tools/#django.test.SimpleTestCase.assertContains).

Reply:
Note that the assertion **should** be in `html=True` mode. I'll fix this.